hawk/src/main/java/com/orhanobut/hawk/Hawk.java

```diff
@@ -97,16 +94,50 @@ public static void init(Context context, LogLevel logLevel) {
      * @param list is the data that will be saved
      */
     public static <T> void put(String key, List<T> list) {
-        if (list == null) {
-            throw new NullPointerException("List<T> may not be null");
+        throwExceptionIf(key == null, "Key cannot be null");
+        throwExceptionIf(list == null, "List<T> may not be null");
+        throwExceptionIf(list.size() == 0, "List<T> may not be null");
```

Comment:
empty list should not throw NPE, this is misleading. (Note that I made the same mistake in the current code, it should be changed as well). It will be best suitable to throw IllegalStateException in this case.

<pre>
if (list.size() == 0) {
    throw new IllegalStateException("List<T>  may not be empty");
}
</pre>